cms/djangoapps/course_creators/admin.py

```diff
@@ -30,26 +33,59 @@ def get_email(obj):
 get_email.short_description = 'email'
 
 
+class CourseCreatorForm(forms.ModelForm):
+    """
+    Admin form for course creator
+    """
+    class Meta:
+        model = CourseCreator
+        fields = '__all__'
+
+    def clean(self):
+        """
+        Validate the 'state', 'organizations' and 'all_orgs' field before saving.
+        """
+        all_orgs = self.cleaned_data.get("all_organizations")
+        orgs = self.cleaned_data.get("organizations").exists()
+        state = self.cleaned_data.get("state")
+        is_all_org_selected_with_orgs = (orgs and all_orgs)
+        is_orgs_added_with_all_orgs_selected = (not orgs and not all_orgs)
+        is_state_granted = state == CourseCreator.GRANTED
+        if is_state_granted:
+            if is_all_org_selected_with_orgs:
+                raise ValidationError(
+                    "The role can be granted either to ALL Organizations or to "
+                    "specific organizations but not both."
```

Comment:
```suggestion
                    "The role can be granted either to ALL organizations or to "
                    "specific organizations but not both."
```
nit: inconsistent capitalization of "organizations"